packages/mirrorful/utils/addToTailwindConfig.ts

```diff
@@ -0,0 +1,314 @@
+          extendIndex,
+          0,
+          tailwindInserts.lineHeight.notExist()
+        )
+        extendIndex++
+      }
+    }
+
+    let mirrorfulImport = ''
+    if (!tailwindFile.includes('.mirrorful/theme_cjs.js')) {
+      mirrorfulImport = `const mirrorful = require('${mirrorfulFolderPath}.mirrorful/theme_cjs.js')\n`
+    }
+    await writeFile(
+      rootPath,
+      mirrorfulImport + tailwindFileArr.join('\n'),
+      'utf-8'
+    )
+  } catch (error) {
+    console.error(error)
+  }
+}
+
+function getExtendThemeIndex(tailwindFileArr: string[]) {
+  let extendIndex = -1,
+    colorsIndex = -1,
+    fontSizeIndex = -1,
+    fontWeightIndex = -1,
+    dropShadowIndex = -1,
+    lineHeightIndex = -1
+
+  for (let i = 0; i < tailwindFileArr.length; i++) {
+    const line = tailwindFileArr[i]
+    if (line.includes('extend:')) {
+      extendIndex = i + 1
+    } else if (line.includes('colors:')) {
+      colorsIndex = i
+    } else if (line.includes('fontSize:')) {
+      fontSizeIndex = i
+    } else if (line.includes('dropShadow:')) {
+      dropShadowIndex = i
+    } else if (line.includes('fontWeight:')) {
+      fontWeightIndex = i
+    } else if (line.includes('lineHeight:')) {
+      lineHeightIndex = i
+    }
+  }
+  return {
+    colorsIndex,
+    fontSizeIndex,
+    dropShadowIndex,
+    extendIndex,
+    fontWeightIndex,
+    lineHeightIndex,
+  }
+}
+
+function doesContainExtendThemes(tailwindFile: string) {
+  const hasColors = tailwindFile.match(/colors:(\s|^\s)(\{|\n)/)
+  const hasFontSizes = tailwindFile.match(/fontSize:(\s|^\s)(\{|\n)/)
+  const hasFontWeights = tailwindFile.match(/fontWeight:(\s|^\s)(\{|\n)/)
+  const hasDropShadow = tailwindFile.match(/dropShadow:(\s|^\s)(\{|\n)/)
+  const hasLineHeight = tailwindFile.match(/lineHeight:(\s|^\s)(\{|\n)/)
+  return {
+    hasColors,
+    hasFontSizes,
+    hasDropShadow,
+    hasFontWeights,
+    hasLineHeight,
+  }
+}
+
+/**
+ *
+ * @param keys The tokens generated by mirrorful
+ * @param path The path to the tailwind file
+ * @returns Boolean
+ */
+async function shouldUpdateTailwindConfig({
+  keys,
+  path,
+}: {
+  keys: Array<TTokens>
+  path: string
+}) {
+  const tailwindFile = await readFile(path, 'utf-8')
+
+  const booleanArr: Record<TTokens, boolean> = {
+    colors: false,
+    boxShadows: false,
+    fontSizes: false,
+    fontWeights: false,
+    lineHeights: false,
+  }
+
+  for (let i = 0; i < keys.length; i++) {
+    const regex = `Tokens.${keys[i]}`
+    booleanArr[keys[i]] = tailwindFile.includes(regex)
+  }
+  return [Object.values(booleanArr).every((a) => a), booleanArr] as const
+}
+
+const skipDirs = [
+  'node_modules',
+  '.next',
+  '.git',
+  '.pnp',
+  'coverage',
+  'out',
+  'build',
+  'lib',
+  '.vercel',
+]
+
+/**
+ *
+ * @param folderName Take in the folder name that needs to be found
+ * @returns Either an empty string or a folder path
+ */
+async function getFolderPath({ folderName }: { folderName: string }) {
```

Comment:
This function is for the case where the user houses the mirrorful project in another folder. So it does a Breadth First Search to try and find it. I thought of this function since in the documentation it says when importing mirrorful it can be from anywhere. The function is also for the case where mirrorful is in an isolated project, due to the pnpm package not working.